astroquery/mast/cloud.py

```diff
@@ -109,30 +109,13 @@ def get_cloud_uri(self, data_product, include_bucket=True, full_url=False):
             found in the cloud, None is returned.
         """
 
-        s3_client = self.boto3.client('s3', config=self.config)
-
-        path = utils.mast_relative_path(data_product["dataURI"])
-        if path is None:
-            raise InvalidQueryError("Malformed data uri {}".format(data_product['dataURI']))
+        uri_list = self.get_cloud_uri_list(data_product, include_bucket=include_bucket, full_url=full_url)
 
-        if 'galex' in path:
-            path = path.lstrip("/mast/")
+        # Making sure we got at least 1 URI from the query above.
+        if len(uri_list) == 0:
+            raise InvalidQueryError("No products found on the cloud for this query.")
```

Comment:
Is this an InvalidQueryError or rather a `NoResultsWarning`? We haven't previously raised the error, but if you think the query has an error in it, rather than just there is no result, then we can raise it from now on (probably would require its own changelog entry)

Reply:
I chose to raise an error rather than a warning so that the output error message is more helpful. Here is what it looks like when I use a `NoResultsWarning`: 

<img width="1071" alt="image" src="https://github.com/astropy/astroquery/assets/32107699/8c966828-0162-4df7-999e-919607176402">

The final `TypeError` is an indirect result of the empty uri list, so it's a vague error message. 

Raising a custom `InvalidQueryError` makes things more clear:

<img width="1072" alt="image" src="https://github.com/astropy/astroquery/assets/32107699/7326177d-c1a7-492a-a666-08252bea29b6">

Reply:
Maybe then a conditional is needed to return the empty result early, before going into the Typeerror. The point is basically to use the InvalidqueryError when there is an issue with the user input. Not having data, imo is not in this category.

Reply:
On second thought, a user will realize that their call to `Observations.get_cloud_uri(product)` returns an empty list before calling `Observations.download_file`, since they have to feed the uri in string form and by default the uri gets returned as a list even when it's 1 URI. So this error handling thing might be redundant.